swiftsimio/initial_conditions/generate_particles.py

```diff
@@ -1,5 +1,1304 @@
+
+    self.tolerance_part: float
+        tolerance for not converged particle fraction: this fraction of
+        particles can be displaced with distances > self.converge_thresh
+
+    self.displ_thresh: float
+        displacement threshold. It's an iteration continuation criterion: Don't
+        stop until every particle is displaced by a distance < `` self.displ_thresh
+        * mean interparticle distance``
+
+    self.delta_init: float or None
+        initial normalization constant for particle motion in units of mean
+        interparticle distance. If ``None`` (default), ``self.delta_init`` will be set
+        such that the maximal displacement found in the first iteration is
+        normalized to 1 mean interparticle distance.
+
+    self.delta_reduct: float
+        normalization constant reduction factor. Intended to be > 0 and < 1.
+        Multiply the normalization constant for particle motion by this factor
+        after every iteration. In certain difficult cases this might help the
+        generation to converge if set to < 1.
+
+    self.delta_min: float
+        minimal normalization constant for particle motion in units of mean
+        interparticle distance.
+
+    self.redist_freq: int
+        particle redistribution frequency.
+        Redistribute a handful of particles every ``self.redistribute_frequency``
+        iteration. How many particles are redistributed is controlled with the
+        ``self.redistribute_frac`` parameter.
+
+    self.redist_frac: float
+        fraction of particles to be redistributed when doing so.
+
+    self.redist_reduct: float
+        multiply the ``self.redistribute_fraction`` parameter by this factor every
+        time particles are being redistributed. In certain difficult cases this
+        might help the generation to converge if set to < 1.
+
+    self.redist_stop: int
+        don't redistribute particles after this many iterations.
+
+    self.dumpfreq: int
+        frequency of dumps of the current state of the iteration. If set to zero,
+        no intermediate results will be stored.
+
+    self.dump_basename: str
+        Basename for intermediate dumps. The filename will be constructed as
+        ``basename + <5 digit zero padded iteration number> + .hdf5``
+
+    self.random_seed: int, optional
+        set a specific random seed
+
+
+
+    Methods
+    ----------
+
+    self.set_random_seed(seed: int)
+        set the random seed to whatever pleases you.
+
+    """
+    # iteration parameters
+    iter_max: int = 10000
+    iter_min: int = 0
+    converge_thresh: float = 1e-4
+    tolerance_part: float = 1e-3
+    displ_thresh: float = 1e-3
+    delta_init: Union[float, None] = None
+    delta_reduct: float = 1.0
+    delta_min: float = 1e-6
+    redist_freq: int = 20
+    redist_frac: float = 0.01
+    redist_reduct: float = 1.0
+    redist_stop: int = 200
+    dump_basename: str = "IC-generation-iteration-"
```

Comment:
Default `dump_basename` should not have dashes in it - `IC_generation_iteration`

Reply:
Done.